scripts/cat/cats.py

```diff
@@ -1030,23 +1029,27 @@ def thoughts(self):
         """ Generates a thought for the cat, which displays on their profile. """
         all_cats = self.all_cats
         other_cat = random.choice(list(all_cats.keys()))
+        game_mode = game.switches['game_mode']
 
         # get other cat
         i = 0
         while other_cat == self.ID and len(all_cats) > 1 or (
-                all_cats.get(other_cat).status in ['kittypet', 'rogue', 'loner']):
+                all_cats.get(other_cat).status in ['kittypet', 'rogue', 'loner', 'former Clancat']):
             other_cat = random.choice(list(all_cats.keys()))
+            # makes sure that a cat won't think about a cat that was dead longer than they've been alive (with 4 moons difference)
+            if all_cats[other_cat].dead and not self.dead and not self.status in ['leader', 'medicine cat']:
+                if (int(all_cats[other_cat].dead_for) + 4) > self.moons:
```

Comment:
I would change it to the way I told you with the relationship stuff.
This would fix the error with the old loner joining a clan fresh and suddenly know freshly died clan cats.

```suggestion
                if all_cats[other_cat].ID not in self.relationships:
```